client/src/components/content/ModelerSidebar/PropertiesPanel/PropertiesPanelComponents/PPParameterInput.jsx

```diff
@@ -1,43 +1,108 @@
-import React from 'react';
-import { Input, Tooltip } from 'antd';
+import React, { useState } from 'react';
+import { Input, Tooltip, Checkbox } from 'antd';
 import { InfoCircleOutlined } from '@ant-design/icons';
-import PropTypes from 'prop-types'
+import PropTypes from 'prop-types';
 import corporateDesign from '../../../../../layout/corporateDesign';
+import styles from '../../ModelerSidebar.module.css';
+import {
+  parameterPropertyStatus,
+  setPropertyForParameter,
+} from '../../../../../utils/attributeAndParamUtils';
 
 /**
  * @description Renders a parameter input field for a given variable
  * @category Client
  * @component
  */
 const PPParameterInput = ({
-    onValueChange,
-    variableName,
-    isRequired,
-    // eslint-disable-next-line no-unused-vars
-    dataType,
-    value
-}) => (
+  onValueChange,
+  variableName,
+  isRequired,
+  // eslint-disable-next-line no-unused-vars
+  dataType,
+  value,
+  robotId,
```

Comment:
As far as I can tell the robotId is passed through from all components simply to be passed in to a method by this component. Please consider loading the robotId from sessionStorage in the method itself, as you are already dealing with sessionStorage items in those methods anyway and it would help us keep the code more clean.

Reply:
Good point. I think it is not really a big problem passing the robotId as a parameter and it has the advantage that the robotId is not dependent on the session storage which is good if we might change the architecture in the future. If you are okay with it I would leave it like that for now and if the passing of properties in general gets too much we can start a refactoring issue for that :)